src/environmentd/src/http.rs

```diff
@@ -282,6 +323,9 @@ impl AuthedClient {
                     if matches!(self.0.session().transaction(), TransactionStatus::Failed(_)) {
                         break;
                     }
+                    if prohibit_stmts(&stmt, &mut results) {
+                        break;
```

Comment:
Is this a break instead of error for a reason? I don't understand what's going on here.

Reply:
`prohibit_stmts` places an error in the result if applicable and returns whether it errored or not. Easy to place the error-generating logic inline here if you find this unclear.

Reply:
My preference is that an unsupported statement anywhere in the entire request should return an error having executed nothing. It's equivalent to a parse error, which occurs when we can't even delimit the string into individual statements.

Reply:
And, thus, this codepath should always be `unreachable!`.

Reply:
Got it; do you want these semantics only for `Simple` or would you extend this prohibition to `Extended` as well?